mock/py/mockbuild/backend.py

```diff
@@ -787,7 +789,15 @@ def get_command(mode):
                 # on re-doing it
                 mode += ['--noprep']
 
-        self.buildroot.doChroot(get_command(mode),
+        # when we used dynamic_buildrequires, we must let rpmbuild check deps
+        # becasue rpmbuild will call %genearte_buildrequires once again
+        # to actually add the BuildRequires to the SRPM metadata
+        # we need to abort the build if %genearte_buildrequires generates a new
+        # unsatisfied dependency
+        # see https://github.com/rpm-software-management/mock/issues/1246
+        # not using --nodeps is safe in this case
+        # becasue the SRPM was generated by the rpm in the chroot
```

Comment:
Do you react on this?  https://github.com/rpm-software-management/mock/blob/c3a4bca61d1863cb8b59e8ba6d2feaff1e67ace7/mock/py/mockbuild/backend.py#L720-L721

Reply:
Yes

Reply:
Well, are you sure that (a) the ability to generate the src.rpm implies that (b) the in-chroot RPM can read the RPM database?

Reply:
Oh. No, I am not sure. Without bootstrap chroot, this might blow up. I'll remove the comment. Whether we can safely do this or not, I ma not sure.

Reply:
I mean -> this is quite OK if bootstrap is turned ON, but we still support non-bootstrapped builds.  Consider that the in-chroot database might be generated by an outdated RPM version from host.

Reply:
I see. In that case, we cannot call `rpmbuild` without `--nodeps` here unless we are in a bootstrap mode.

A hacky solution for disabled bootstrap is to call installSrpmDeps on the final built SRPM and die if it fails to install something. That's not as nice.

Alternatively, we might force the `--nodeps`  when not in bootstrap and claim this problem is only fixed when bootstrap is on. What do you think?

Reply:
I pushed a fixup commit that only does this when using a bootstrap chroot. It's not perfect, but it is a good enough solution for Fedora.